include/hipSYCL/sycl/libkernel/sscp/builtins.hpp

```diff
@@ -405,6 +405,28 @@ HIPSYCL_BUILTIN T __hipsycl_clamp(T x, T minval, T maxval) noexcept {
     sscp_builtins::__hipsycl_max(x, minval), maxval);
 }
 
+
+
+template<class T,std::enable_if_t<(std::is_same_v<T, __hipsycl_uint8> || std::is_same_v<T, __hipsycl_int8>), int> = 0>
+HIPSYCL_BUILTIN T __hipsycl_clz(T x) noexcept {
+  return __hipsycl_sscp_clz_u8(x);
```

Comment:
Your sscp builtin implementations operate on and return unsigned types in all cases. So I don't think you can treat them in the same code path, as you would need `static_cast` to convert between them.

Also, `__hipsycl_*` types are mainly intended for low-level stuff that enters our bitcode libraries. This stuff is still regular C++, so you can use normal C++ types here :)

Reply:
I've change it to :

```c++
template <class T,
          std::enable_if_t<
              (std::is_same_v<T, unsigned int> || std::is_same_v<T, int> ||
               std::is_same_v<T, unsigned short> || std::is_same_v<T, short> ||
               std::is_same_v<T, unsigned char> ||
               std::is_same_v<T, signed char> || std::is_same_v<T, char>),
              int> = 0>
HIPSYCL_BUILTIN T __hipsycl_clz(T x) noexcept {
  return __hipsycl_sscp_clz_u32(static_cast<__hipsycl_uint32>(x));
}

template <class T, std::enable_if_t<(std::is_same_v<T, unsigned long long> ||
                                     std::is_same_v<T, long long>),
                                    int> = 0>
HIPSYCL_BUILTIN T __hipsycl_clz(T x) noexcept {
  return __hipsycl_sscp_clz_u64(static_cast<__hipsycl_uint64>(x));
}
```